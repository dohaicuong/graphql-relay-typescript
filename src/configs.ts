export const PORT = process.env.PORT || 4000
export const SERVICE_NAME = process.env.SERVICE_NAME || 'graphql-service'