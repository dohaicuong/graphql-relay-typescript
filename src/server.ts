import { ApolloServer } from 'apollo-server'

import schema from './modules'

import { PrismaClient } from '@prisma/client'
export const prisma = new PrismaClient()

import createLogPlugin from './plugins/logs'
import winston, { format } from 'winston'
import { SERVICE_NAME } from './configs'
const logger = winston.createLogger({
  level: 'error',
  format: format.combine(
    format.colorize(),
    format.prettyPrint(),
    format.splat(),
    format.simple(),
    format.label({ label: SERVICE_NAME }),
    format.timestamp(),
    format.printf(({ timestamp, label, level, message }) => {
      // @ts-ignore
      return `${timestamp} [${label}] ${level}: ${message.operation} - ${JSON.stringify(message.info, null, 4)}`
    })
  ),
  transports: [
    // new winston.transports.File({ filename: 'combined.log' })
    new winston.transports.Console(),
  ]
})

export type Context = {
  req: any
  logger: winston.Logger
  prisma: PrismaClient
}

export const options = {
  schema,
  context: ({ req }: any) => {
    return {
      req,
      logger,
      prisma
    }
  },
  plugins: [
    createLogPlugin(logger)
  ]
}

export default new ApolloServer(options)