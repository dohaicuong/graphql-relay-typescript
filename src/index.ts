import server, { prisma } from './server'
import { PORT } from './configs'

server
  .listen({ port: PORT })
  .then(({ url }) => console.log(url))
  .finally(() => {
    prisma.disconnect()
  })