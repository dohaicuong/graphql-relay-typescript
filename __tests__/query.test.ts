import { createTestClient } from 'apollo-server-testing'
import { ApolloServer } from 'apollo-server'
import { options } from '../src/server'

it('fetch a single post', async () => {
  const mockPrisma = {
    post: {
      findOne: () => ({ id: 1, title: 'dkm', content: 'dkm too' })
    }
  }

  const server = new ApolloServer({
    ...options,
    context: {
      ...options.context,
      prisma: mockPrisma
    }
  })

  const { query } = createTestClient(server)

  const res = await query({
    query: `
      query PostQuery($id: ID!) {
        post(id: $id) {
          id
          title
          content
        }
      }
    `,
    variables: { id: 1 }
  })
  expect(res).toMatchSnapshot()
})