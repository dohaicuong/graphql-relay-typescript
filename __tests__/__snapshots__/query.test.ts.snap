// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch a single post 1`] = `
Object {
  "data": Object {
    "post": Object {
      "content": "dkm too",
      "id": "1",
      "title": "dkm",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
